### PR TITLE
Add name filter to API for GetMilestoneList

### LINKED ISSUE #12260: API: extend Get Milestone List to filter by name
extend https://try.gitea.io/api/swagger#/issue/issueGetMilestonesList to filter by name
this is usefull if you have the mile-Name and want to get the ID, ...

### PR DESCRIPTION
**Pull request by 6543**
it include also a refactor (merge GetMilestones and GetmilestonesByRepoID)

close #12260

neded for https://gitea.com/gitea/go-sdk/issues/383 and https://gitea.com/gitea/tea/pulls/149

### CODE REVIEW

File: models/issue_milestone.go [hunk 1]
```diff
@@ -330,41 +330,46 @@ func (milestones MilestoneList) getMilestoneIDs() []int64 {
...
+	State    api.StateType
+	Name     string
+	SortType string
+}
+
+// GetMilestones returns milestones filtered by GetMilestonesOption's
+func GetMilestones(opts GetMilestonesOption) (MilestoneList, error) {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
```
Comment (lunny) [inline models/issue_milestone.go]: We don't need a transaction here.

**Review by lunny — CHANGES_REQUESTED** [review]
Otherwise LGTM

**Comment by 6543** [comment]
@lunny done

**Comment by 6543** [comment]
need labels :)

**Review by lunny — APPROVED** [review]

**Review by zeripath — APPROVED** [review]
I note you've changed the default to do a sort by deadline then by id - we're probably going to have to update our API to allow this kind of multiple sorting at some point.

**Pull request merged**